inc/saitwamp.h

```diff
@@ -0,0 +1,808 @@
+
+} sai_twamp_session_role_t;
+
+/**
+ * @brief SAI Two-Way Active Measurement Protocol mode
+ */
+typedef enum _sai_twamp_mode_t
+{
+    /**
+     * @brief Means TWAMP protocol when enabling Two-Way Active Measurement Protocol full mode
+     */
+    SAI_TWAMP_MODE_FULL = 0,
+
+    /**
+     * @brief Means TWAMP Light protocol when enabling Two-Way Active Measurement Protocol light mode
+     */
+    SAI_TWAMP_MODE_LIGHT
+
+} sai_twamp_mode_t;
+
+/**
+ * @brief SAI Two-Way Active Measurement transmitting mode
+ */
+typedef enum _sai_twamp_pkt_tx_mode_t
+{
+    /** @brief Continue to send Two-Way Active Measurement Protocol test packet */
+    SAI_TWAMP_PKT_TX_MODE_CONTINUOUS = 0,
+
+    /** @brief Only send Two-Way Active Measurement Protocol test packets with assigned number */
+    SAI_TWAMP_PKT_TX_MODE_PACKET_COUNT,
+
+    /** @brief Send Two-Way Active Measurement Protocol test packets during period time */
+    SAI_TWAMP_PKT_TX_MODE_PERIOD
+
+} sai_twamp_pkt_tx_mode_t;
+
+/**
+ * @brief SAI Two-Way Active Measurement format of timestamp
+ */
+typedef enum _sai_twamp_timestamp_format_t
+{
+    /**
+     * @brief Packet timestamp format is Network Time Protocol format, 32 bit second and 32 bit fractional part of second
+     */
+    SAI_TWAMP_TIMESTAMP_FORMAT_NTP = 0,
+
+    /**
+     * @brief Packet timestamp format is PTP format, 32 bit second and 32 bit nanosecond
+     */
+    SAI_TWAMP_TIMESTAMP_FORMAT_PTP
+
+} sai_twamp_timestamp_format_t;
+
+/**
+ * @brief SAI Two-Way Active Measurement Protocol type of encapsulation
+ */
+typedef enum _sai_twamp_encapsulation_type_t
+{
+    /**
+     * @brief IP Encapsulation, L2 header | IP(v4/v6) header | UDP header | Two-Way Active Measurement Protocol test packet
+     */
+    SAI_TWAMP_ENCAPSULATION_TYPE_IP = 0,
+
+    /**
+     * @brief L2 Virtual Private Network Encapsulation, L2 header | MPLS Label List | L2 header | IP(v4/v6) header | UDP header | Two-Way Active Measurement Protocol test packet
+     */
+    SAI_TWAMP_ENCAPSULATION_TYPE_MPLS_L2VPN,
+
+    /**
+     * @brief L3 Virtual Private Network Encapsulation, L2 header | MPLS Label List | IP(v4/v6) header | UDP header | Two-Way Active Measurement Protocol test packet
+     */
+    SAI_TWAMP_ENCAPSULATION_TYPE_MPLS_L3VPN,
+
+    /**
+     * @brief VXLAN Network Encapsulation, L2 header | IP(v4/v6) header | UDP header | VXLAN header | L2 header | IP(v4/v6) header | UDP header | Two-Way Active Measurement Protocol test packet
+     */
+    SAI_TWAMP_ENCAPSULATION_TYPE_VXLAN
+
+} sai_twamp_encapsulation_type_t;
+
+/**
+ * @brief SAI attributes for Two-Way Active Measurement Protocol session
+ */
+typedef enum _sai_twamp_session_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_TWAMP_SESSION_ATTR_START,
+
+    /**
+     * @brief Two-Way Active Measurement Protocol mode: light mode and full mode
+     *
+     * @type sai_twamp_mode_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
```

Comment:
Thanks for comments. 
TWAMP Light is simple version of TWAMP. When you create a twamp session, you must configure the attribute as TWAMP or TWAMP Light, so there is no default value and it is mandatory. We provide this attribute because it may support TWAMP feature in the furture.

Reply:
It is different between TWAMP and TWAMP Light for configuration.